From 77b5168af1f4bbcd989f73a77d94455921241b85 Mon Sep 17 00:00:00 2001
From: Frederic Lefebvre <frederic.lefebvre@qt.io>
Date: Fri, 16 Oct 2026 10:55:54 +0200
Subject: [PATCH] Wayland client: pass activeWindowFocusReason on window
 activations

QWaylandDisplay::handleWaylandSync called
QWindowSystemInterface::handleFocusWindowChanged without a focus reason,
so the default Qt::OtherFocusReason was used. As a result, items
receiving focus through window activation reported OtherFocusReason.

Pass Qt::ActiveWindowFocusReason explicitly, matching the xcb plugin.

Change-Id: Ia5b1fff7566b3b351ad587977d84566eed72b43b
Reviewed-by: David Edmundson <davidedmundson@kde.org>
(cherry picked from commit 41021bcc08cfef5178b9f6e83e087b48db5577fc)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
Origin: upstream, https://invent.kde.org/qt/qt/qtbase/-/commit/77b5168af1f4bbcd989f73a77d94455921241b85
Forwarded: not-needed
Last-Update: 2026-10-29
---
 src/plugins/platforms/wayland/qwaylanddisplay.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/wayland/qwaylanddisplay.cpp b/src/plugins/platforms/wayland/qwaylanddisplay.cpp
index ce21f378ca4..a14c553765f 100644
--- a/src/plugins/platforms/wayland/qwaylanddisplay.cpp
+++ b/src/plugins/platforms/wayland/qwaylanddisplay.cpp
@@ -1082,7 +1082,7 @@ void QWaylandDisplay::handleWaylandSync()
     if (mActiveSubSurface)
         activeWindow = mActiveSubSurface->window();
     if (activeWindow != QGuiApplication::focusWindow())
-        QWindowSystemInterface::handleFocusWindowChanged(activeWindow);
+        QWindowSystemInterface::handleFocusWindowChanged(activeWindow, Qt::ActiveWindowFocusReason);
 }
 
 const wl_callback_listener QWaylandDisplay::syncCallbackListener = {
